pkg/virt-controller/watch/migration.go

```diff
@@ -837,7 +837,8 @@ func (c *MigrationController) handleMarkMigrationFailedOnVMI(migration *virtv1.V
 		return err
 	}
 	log.Log.Object(vmi).Infof("Marked Migration %s/%s failed on vmi due to target pod disappearing before migration kicked off.", migration.Namespace, migration.Name)
-	c.recorder.Event(vmi, k8sv1.EventTypeWarning, FailedMigrationReason, fmt.Sprintf("VirtualMachineInstance migration uid %s failed. reason: target pod is down", string(migration.UID)))
+	vmiCopy.Status.MigrationState.FailureReason = "Target pod is down"
+	c.recorder.Event(vmi, k8sv1.EventTypeWarning, FailedMigrationReason, fmt.Sprintf("VirtualMachineInstance migration uid %s failed. reason: %s", string(migration.UID), vmiCopy.Status.MigrationState.FailureReason))
```

Comment:
Is there a way that we will already have a Failure Reason from virt-handler at this point?

Reply:
Yeah that's possible. What do you suggest?

Reply:
Wouldn't creating the event for the migration object solve this?

Reply:
@Barakmor1 I think these are two separate records. We need to store this info in the status for convenience. That way everything related to this migration is stored in one place.

Reply:
> Yeah that's possible. What do you suggest?

@jean-edouard I think we can merge. To simply check whether `MigrationState.FailureReason` has something already and add this message. wdyt?